### PR TITLE
VAGOV-megamenu: add e2e tests for launch of data-driven megamenu

### PR DESCRIPTION
**Pull request by aurora-a-k-a-lightning**
## Description
adds a few e2e tests for the data-driven megamenu. it mostly checks for link counts and header title text

## Testing done
e2e tests

## Screenshots
image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/plugins/create-header-footer.js [hunk 1]
```diff
@@ -97,22 +90,6 @@ function createHeaderFooterData(buildOptions) {
...
-        // menu structure while we launch and QA the Drupal-powered
-        // megaMenu header. The assertion should be removed once
-        // we are confident that Drupal reliably generates the menu data.
-        assert.deepStrictEqual(
-          megaMenuFromDrupal,
-          megaMenuFromVagovContent,
-          'The Drupal data aligns with that from vagov-content.',
-        );
-      }
-
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-header-footer.js]: Since we have an E2E test in its place, this assertion can be removed.

**Review by ncksllvn — APPROVED** [review]
LGTM

**Comment by accbjt** [comment]
Can we make sure we have an axeCheck test on this.

**Review by accbjt — APPROVED** [review]
Please make sure we have an axeCheck on this, otherwise everything else looks good.

**Review by cvalarida — APPROVED** [review]
Looks great! I ran it locally, changed a link count assertion and it broke as expected.

**Pull request merged**